### PR TITLE
ENGG-1600: Templates list UI refactor

### PR DESCRIPTION
**Pull request by RuntimeTerror10**
Screenshot 2024-04-24 at 9 24 10 AM

Closes issue: 

## ✅ Checklist:

- [ ] Make sure linting and unit tests pass.
- [ ] No install/build warnings introduced.
- [ ] Verified UI in browser.
- [ ] For UI changes, added/updated analytics events (if applicable).
- [ ] For changes in extension's code, manually tested in Chrome and Firefox.
- [ ] For changes in extension's code, both MV2 and MV3 are covered.
- [ ] Added/updated unit tests for this change.
- [ ] Raised pull request to update corresponding documentation (if already exists).

### CODE REVIEW

File: app/src/features/rules/screens/templatesList/components/TemplatesTable/TemplatesTable.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    <>
+      {isTemplatePreviewModalVisible && (
+        <TemplatePreviewModal
+          rule={templateToPreview}
+          isOpen={isTemplatePreviewModalVisible}
+          toggle={() => setIsTemplatePreviewModalVisible(false)}
+          source={SOURCE.TEMPLATES_SCREEN}
+        />
+      )}
+      <div className="rq-templates-table">
```
Comment (wrongsahil) [inline app/src/features/rules/screens/templatesList/components/TemplatesTable/TemplatesTable.tsx:32]: Add a Header with search here too for consistency

**Review by wrongsahil — CHANGES_REQUESTED** [review]

File: app/src/componentsV2/BreadCrumb/BreadCrumb.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    <div className="rq-breadcrumb">
+      {items.map((item, index) => (
+        <React.Fragment key={index}>
+          <span className="rq-breadcrumb-item">{item}</span>
+          {index !== items.length - 1 && " > "}
+        </React.Fragment>
+      ))}
+    </div>
+  );
+};
```
Comment (wrongsahil) [inline app/src/componentsV2/BreadCrumb/BreadCrumb.tsx:19]: Don't make a common component right now. Breadcrumb logic should be determined by routing and we'll be making them clickable too in future. So this won't scale much. 

Lets duplicate it everywhere for now

File: app/src/features/rules/screens/templatesList/components/TemplatesTable/TemplatesTable.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        <ContentListTable
+          id="templates-list-table"
+          size="small"
+          columns={tableColumns}
+          data={filteredRecords}
+          locale={{
+            emptyText: <Empty image={Empty.PRESENTED_IMAGE_SIMPLE} description="No Templates found" />,
+          }}
+          scroll={{ y: `calc(100vh - 232px)` }}
```
Comment (wrongsahil) [inline app/src/features/rules/screens/templatesList/components/TemplatesTable/TemplatesTable.tsx:50]: What is this scroll for?

File: app/src/features/rules/screens/templatesList/components/TemplatesTable/hooks/useTemplatesTableColumns.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        }
+        return (
+          <>
+            <div className="template-name" onClick={() => handlePreviewTemplate(record)}>
+              {record.name}
+            </div>
+            <div className="template-rule-tags">
+              {ruleTypes.map((ruleType, index) => (
+                <RuleTypeTag ruleType={ruleType} title={ruleType.toUpperCase()} />
+              ))}
```
Comment (wrongsahil) [inline app/src/features/rules/screens/templatesList/components/TemplatesTable/hooks/useTemplatesTableColumns.tsx:54]: We used to have2 columns for this. Why did we move this to single column?
Right now it is taking 2 rows which is taking too much space. Can we keep rule type in a separate column only?

File: app/src/features/rules/screens/templatesList/modals/TemplatePreviewModal/TemplatePreviewModal.tsx [hunk 5]
```diff
@@ -1,23 +1,31 @@
 import React from "react";
 import { useDispatch, useSelector } from "react-redux";
 import { useNavigate } from "react-router-dom";
```
Comment (wrongsahil) [inline app/src/features/rules/screens/templatesList/modals/TemplatePreviewModal/TemplatePreviewModal.tsx:3]: Are we not using same modal in SharedList Rule viewer?

**Review by wrongsahil — CHANGES_REQUESTED** [review]

File: app/src/features/rules/screens/templatesList/modals/TemplatePreviewModal/TemplatePreviewModal.tsx (hunk 5 above)
Comment (RuntimeTerror10) [reply app/src/features/rules/screens/templatesList/modals/TemplatePreviewModal/TemplatePreviewModal.tsx:3]: No SharedList rule viewer modal is different

File: app/src/features/rules/screens/templatesList/components/TemplatesTable/TemplatesTable.tsx (hunk 3 above)
Comment (RuntimeTerror10) [reply app/src/features/rules/screens/templatesList/components/TemplatesTable/TemplatesTable.tsx:50]: This is the scroll for table body

Comment (wrongsahil) [reply app/src/features/rules/screens/templatesList/components/TemplatesTable/TemplatesTable.tsx:50]: Why do we need to define this explicitly though?

**Review by wrongsahil — APPROVED** [review]

**Pull request merged**